ios/browser/api/ads/brave_ads.mm

```diff
@@ -212,6 +212,15 @@ - (void)notifyBraveNewsIsEnabledPreferenceDidChange:(BOOL)isEnabled {
                  value:base::Value(isEnabled)];
 }
 
+- (void)notifySponsoredImagesIsEnabledPreferenceDidChange:(BOOL)isEnabled {
+  [self setProfilePref:ntp_background_images::prefs::
+                           kNewTabPageShowBackgroundImage
+                 value:base::Value(isEnabled)];
+  [self setProfilePref:ntp_background_images::prefs::
+                           kNewTabPageShowSponsoredImagesBackgroundImage
+                 value:base::Value(isEnabled)];
+}
```

Comment:
These prefs aren't registered on iOS since they only get registered via the ViewCounterService which isn't compiled/called into, thus will lead to a crash on set/access

Reply:
Fixed, thanks